logstreams/src/test/java/io/zeebe/logstreams/state/ReplicateSnapshotControllerTest.java

```diff
@@ -0,0 +1,137 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.logstreams.state;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.groups.Tuple.tuple;
+
+import io.zeebe.db.impl.DefaultColumnFamily;
+import io.zeebe.db.impl.rocksdb.ZeebeRocksDbFactory;
+import io.zeebe.logstreams.processor.SnapshotChunk;
+import io.zeebe.logstreams.processor.SnapshotReplication;
+import io.zeebe.logstreams.util.RocksDBWrapper;
+import io.zeebe.test.util.AutoCloseableRule;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class ReplicateSnapshotControllerTest {
+  @Rule public TemporaryFolder tempFolderRule = new TemporaryFolder();
+  @Rule public AutoCloseableRule autoCloseableRule = new AutoCloseableRule();
+
+  private StateSnapshotController replicatorSnapshotController;
+  private StateSnapshotController receiverSnapshotController;
+  private Replicator replicator;
+
+  @Before
+  public void setup() throws IOException {
+    final File runtimeDirectory = tempFolderRule.newFolder("runtime");
+    final File snapshotsDirectory = tempFolderRule.newFolder("snapshots");
+    final DataStorage storage = new DataStorage(runtimeDirectory, snapshotsDirectory);
+
+    final File receiverRuntimeDirectory = tempFolderRule.newFolder("runtime-receiver");
+    final File receiverSnapshotsDirectory = tempFolderRule.newFolder("snapshots-receiver");
+    final DataStorage receiverStorage =
+        new DataStorage(receiverRuntimeDirectory, receiverSnapshotsDirectory);
+
+    replicator = new Replicator();
+    replicatorSnapshotController =
+        new StateSnapshotController(
+            storage, replicator, ZeebeRocksDbFactory.newFactory(DefaultColumnFamily.class));
+    receiverSnapshotController =
+        new StateSnapshotController(
+            receiverStorage, replicator, ZeebeRocksDbFactory.newFactory(DefaultColumnFamily.class));
+
+    autoCloseableRule.manage(replicatorSnapshotController);
+    autoCloseableRule.manage(receiverSnapshotController);
+
+    final String key = "test";
+    final int value = 0xCAFE;
+    final RocksDBWrapper wrapper = new RocksDBWrapper();
+    wrapper.wrap(replicatorSnapshotController.openDb());
+    wrapper.putInt(key, value);
+  }
+
+  @Test
+  public void shouldReplicateSnapshotChunks() {
+    // given
+    replicatorSnapshotController.takeSnapshot(1);
+
+    // when
+    replicatorSnapshotController.replicateLatestSnapshot();
+
+    // then
+    final List<SnapshotChunk> replicatedChunks = replicator.replicatedChunks;
+    final int totalCount = replicatedChunks.size();
+    assertThat(totalCount).isGreaterThan(0);
```

Comment:
Please verify that it replicates all files of the snapshot.

Reply:
https://github.com/zeebe-io/zeebe/pull/2356/files/dbccb868cf650a93afb9fb267a26fa8fa1cca617#diff-9800c62fb09350d6721dd5473418b415R89